code/datums/components/cell_component.dm

```diff
@@ -0,0 +1,195 @@
+
+	if(inserted_cell)
+		QDEL_NULL(inserted_cell)
+		inserted_cell = null
+	return ..()
+
+/**
+ * The basic way of processing the cell, with included feedback.
+ *
+ * This proc is the basic way of processing the cell, with included feedback.
+ * It will return a bitflag if it failed to use the power, or COMPONENT_POWER_SUCCESS if it succeeds.
+ * Arguments:
+ * * use_amount - an override
+ * * check_only - will only return if it can use the cell and feedback relating to that including any relevant detail
+ */
+/datum/component/cell/proc/simple_power_use(datum/source, use_amount, mob/user, check_only)
+	SIGNAL_HANDLER
+
+	if(!use_amount)
+		use_amount = power_use_amount
+
+	if(!inserted_cell)
+		if(user)
+			to_chat(user, span_danger("There is no cell inside [equipment]"))
+		return COMPONENT_NO_CELL
+
+	if(check_only && inserted_cell.charge < use_amount)
+		if(user)
+			to_chat(user, span_danger("The cell inside [equipment] does not have enough charge to perform this action!"))
+		return COMPONENT_NO_CHARGE
+
+	if(!inserted_cell.use(use_amount))
+		inserted_cell.update_appearance()  //Updates the attached cell sprite - Why does this not happen in cell.use?
+		if(user)
+			to_chat(user, span_danger("The cell inside [equipment] does not have enough charge to perform this action!"))
+		return COMPONENT_NO_CHARGE
+
+	inserted_cell.update_appearance()
+
+	return COMPONENT_POWER_SUCCESS
+
+/datum/component/cell/proc/examine_cell(atom/A, mob/user, list/examine_list)
+	SIGNAL_HANDLER
+
+	if(!inserted_cell)
+		examine_list += span_danger("It does not have a cell inserted!")
+	else
+		examine_list += span_notice("It has a [inserted_cell] inserted. \
+						The cell has <b>[inserted_cell.percent()]%</b> charge remaining. \
+						<b>Alt-click</b> to remove the cell.")
+
+/// Handling of cell removal.
+/datum/component/cell/proc/remove_cell(datum/source, mob/user)
+	SIGNAL_HANDLER
+	if(!equipment.can_interact(user))
+		return
+
+	if(!cell_can_be_removed)
+		return
+
+	if(!isliving(user))
+		return
+
+	if(inserted_cell)
+		to_chat(user, span_notice("You remove [inserted_cell] from [equipment]."))
+		playsound(equipment, 'sound/weapons/magout.ogg', 40, TRUE)
+		inserted_cell.forceMove(get_turf(equipment))
+		INVOKE_ASYNC(user, TYPE_PROC_REF(/mob/living, put_in_hands), inserted_cell)
+		inserted_cell = null
+		if(on_cell_removed)
+			on_cell_removed.Invoke()
+		handle_cell_overlays(TRUE)
+	else
+		to_chat(user, span_danger("There is no cell in [equipment]!"))
+
+/// Handling of cell insertion.
+/datum/component/cell/proc/insert_cell(datum/source, obj/item/inserting_item, mob/living/user, params)
+	SIGNAL_HANDLER
+	if(!equipment.can_interact(user))
+		return
+
+	if(!istype(inserting_item, /obj/item/stock_parts/cell))
+		return
+
+	if(inserted_cell) //No quickswap compatibility
+		to_chat(user, span_danger("There is already a cell in [equipment]!"))
+		return
+
+	to_chat(user, span_notice("You connect [inserting_item] onto [equipment]."))
```

Comment:
```suggestion
	to_chat(user, span_notice("You add [inserting_item] to [equipment]."))
```

Reply:
Maybe not exactly this, right now it does
`you connect the cell onto the electric welder`
maybe `you connect the cell to the electric welder` will be what i'll do

Reply:
I was considering that too yeah. works